src/xpk/core/cluster.py

```diff
@@ -729,78 +729,48 @@ def update_cluster_with_gcsfuse_driver_if_necessary(args) -> int:
   return 0
 
 
-def test_and_retry_credentials_with_dns_logic(args) -> int:
-  """Tests kubectl credentials and retries with default settings if a DNS error is found.
+def get_cluster_credentials(args) -> int:
+  """Run cluster configuration command to set the kubectl config.
 
   Args:
     args: user provided arguments for running the command.
 
   Returns:
-    0 if credentials are valid after retrying, 1 otherwise.
+    0 if successful and 1 otherwise.
   """
+  location = get_cluster_location(args.project, args.cluster, args.zone)
 
-  xpk_print('Testing credentials with kubectl...')
-  kubectl_command = 'kubectl get pods'
-  kubectl_return_code, kubectl_output = run_command_for_value(
-      kubectl_command, 'kubectl get pods'
-  )
-  if kubectl_return_code == 0:
-    xpk_print('Credentials test succeeded.')
-    return 0
+  _get_credentials(args.project, args.cluster, location, dns_endpoint=True)
 
-  dns_endpoint_error = (
-      'control_plane_endpoints_config.dns_endpoint_config.allow_external_traffic'
-      ' is disabled'
-  )
-  if dns_endpoint_error not in kubectl_output:
-    xpk_print(f'kubectl failed with an unhandled error: {kubectl_output}')
-    xpk_exit(kubectl_return_code)
-  xpk_print(
-      'Detected DNS endpoint-related error. Retrying without --dns-endpoint'
-      ' flag...'
-  )
+  xpk_print('Testing credentials with kubectl...')
```

Comment:
nit: Let's remove this print, and change the task name from `'kubectl get pods'` to `Test credentials with kubectl`?